### PR TITLE
add unique idx on user column in mobileapp authtoken table

### PR DESCRIPTION
**Pull request by joeyorlando**
# Which issue(s) this PR fixes
Solves the (rare) issue where a user could potentially have > 1 mobileapp auth token, leading to 500 errors when trying to interact w/ the authtoken (ex. disconnect a mobile app from a user's profile):
```shell
2023-03-07 10:12:13 source=engine:app google_trace_id=e14bf933d634068a48caf093ce43c7f5/5550677047491218352 logger=django.request Internal Server Error: /api/internal/v1/users/U6WJ3BRLM1TR7/unlink_backend
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/etc/app/apps/api/views/user.py", line 453, in unlink_backend
    backend.unlink_user(user)
  File "/etc/app/apps/mobile_app/backend.py", line 34, in unlink_user
    token = MobileAppAuthToken.objects.get(user=user)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 439, in get
    raise self.model.MultipleObjectsReturned(
apps.mobile_app.models.MobileAppAuthToken.MultipleObjectsReturned: get() returned more than one MobileAppAuthToken -- it returned 2!
```

## Checklist

- [x] Tests updated
- [ ] Documentation added (N/A)
- [x] `CHANGELOG.md` updated

### CODE REVIEW

File: engine/apps/mobile_app/models.py [hunk 1]
```diff
@@ -51,9 +51,7 @@ def create_auth_token(cls, user: User, organization: Organization) -> Tuple["Mob
 
 
 class MobileAppAuthToken(BaseAuthToken):
-    user = models.ForeignKey(
-        to=User, null=False, blank=False, related_name="mobile_app_auth_tokens", on_delete=models.CASCADE
-    )
+    user = models.OneToOneField(to=User, null=False, blank=False, on_delete=models.CASCADE)
```
Comment (joeyorlando) [inline engine/apps/mobile_app/models.py:54]: main focus of the PR. Rest is related to new API endpoint tests I added.

Comment (iskhakov) [reply engine/apps/mobile_app/models.py:54]: do we have one to many relations already?
Do we need any data migration/cleanup?

Comment (joeyorlando) [reply engine/apps/mobile_app/models.py:54]: I've checked all of our environments and we're all good, no migration needed.

We only had one instance of this pop up (which affected @imtoori), this was manually fixed.

**Review by vadimkerr — APPROVED** [review]
LGTM 👍

File: engine/apps/mobile_app/models.py (hunk 1 above)
Comment (vadimkerr) [reply engine/apps/mobile_app/models.py:54]: What about OSS instances, I guess it's possible that some duplicates are created on some of the OSS stacks?

File: engine/apps/mobile_app/migrations/0002_alter_mobileappauthtoken_user.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('user_management', '0008_organization_is_grafana_incident_enabled'),
+        ('mobile_app', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AlterField(
```
Comment (vadimkerr) [inline engine/apps/mobile_app/migrations/0002_alter_mobileappauthtoken_user.py:15]: Minor suggestion: consider a `RunPython` operation that deletes duplicates before running `AlterField`

Comment (joeyorlando) [reply engine/apps/mobile_app/migrations/0002_alter_mobileappauthtoken_user.py:24]: Good idea 👍 addressed in [11376e5](https://github.com/grafana/oncall/pull/1482/commits/11376e57e3afeeb2e166ea9d42a8b2cfd21b2dcf).

**Before**
Screenshot 2023-03-08 at 13 40 02

**After**
Screenshot 2023-03-08 at 13 40 55

File: engine/apps/mobile_app/models.py (hunk 1 above)
Comment (joeyorlando) [reply engine/apps/mobile_app/models.py:54]: see [comment above](https://github.com/grafana/oncall/pull/1482/files#r1129372892)

**Pull request merged**